devtools/src/arcs-bug-report.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {PolymerElement} from '../deps/@polymer/polymer/polymer-element.js';
+import {html} from '../deps/@polymer/polymer/lib/utils/html-tag.js';
+import {MessengerMixin} from './arcs-shared.js';
+import '../deps/jszip/dist/jszip.js';
+
+/**
+ * Saves/loads a bug report which contains debug messages sent to devtools.
+ */
+class ArcsBugReport extends MessengerMixin(PolymerElement) {
+  static get template() {
+    return html`
+    <style include="shared-styles">
+      :host {
+        display: flex;
+        flex-direction: column;
+      }
+    </style>
+    <div class="wrapper">
+      <input type="file" id="fileElem" style="display:none" on-change="_loadBugReport">
+      <button id="fileSelect">Load Bug Report</button>
+      <button type="button" on-click="_saveBugReport">Save Bug Report</button>
+    </div>`;
+  }
+
+  static get is() { return 'arcs-bug-report'; }
+
+  constructor() {
+    super();
+    this.messages = [];
+  }
+
+  ready() {
+    super.ready();
+    const fileElem = this.$.fileElem;
+    this.$.fileSelect.addEventListener("click", function (e) {
+      if (fileElem) {
+        fileElem.click();
+      }
+    }, false);
+  }
+
+  onRawMessageBundle(messages) {
+    this.messages = this.messages.concat(messages);
+  }
+
+  _saveBugReport() {
+    var zip = new JSZip();
+    var file = zip.file("bugreport.txt", JSON.stringify(this.messages));
```

Comment:
It doesn't appear like JSZip is compressing the file. The zip has the same size (actually, is slightly larger) that the txt file. But if I unzip it and zip it locally on my machine I went from 2.1MB to 216KB...

I've looked around and found this: https://github.com/Stuk/jszip/issues/406

Reply:
Adding that option fixes the issue.